notebooks/refacto.ipynb

```diff
@@ -0,0 +1,1822 @@
+        "        raise Exception (\"Hyperparameter tuning!!!\")\n",
+        "       \n",
+        "\n",
+        "    def train_model(self):\n",
+        "        \"\"\"\n",
+        "        Train model with best parameters.\n",
+        "        \"\"\"\n",
+        "        self.logger.error(\"Training failed!!!\")\n",
+        "        raise Exception (\"Training fail!!!\")\n",
+        "        \n",
+        "\n",
+        "    def run(self):\n",
+        "        \"\"\" \n",
+        "        Run class methods in order.\n",
+        "        \"\"\"\n",
+        "        self.logger.info('Running Model...')\n",
+        "        self.load_dataset()\n",
+        "        self.preprocessing()\n",
+        "        self.train_test_split()\n",
+        "        self.parameter_tuning()\n",
+        "        self.train_model()\n",
+        "        self.logger.info('Modeling finished!!!')"
+      ],
+      "execution_count": 64,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "time: 56.3 ms (started: 2021-06-02 14:52:27 +00:00)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IHqed0zDCCIm"
+      },
+      "source": [
+        "# Baseline (Mean)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "f_35JxWrCFHf",
+        "outputId": "6feedd63-2d23-4ee5-c1ac-e543b7f6c27d"
+      },
+      "source": [
+        "class Baseline(Model):\n",
+        "    \"\"\"\n",
+        "    Baseline (mean) pipieline.\n",
+        "    \"\"\"\n",
+        "    def __init__(self, params):\n",
+        "        \"\"\"\n",
+        "        Init pipeline with datasets, model parameters and performance measurements. \n",
+        "        \"\"\"\n",
+        "        Model.__init__(self, params)\n",
+        "        self.model_name = \"decision_trees\"\n",
+        "\n",
+        "        self.classes = ['A','B','C','D','E','F','G']\n",
+        "        self.y_test_class = None\n",
+        "        self.y_pred = None # Consommation energie\n",
+        "        self.y_pred_class = None # Classe consommation energie\n",
+        "\n",
+        "        self.mean = None\n",
+        "        self.confusion_matrix = None\n",
+        "        self.classification_metrics = None \n",
+        "\n",
+        "        self.run() \n",
+        "\n",
+        "\n",
+        "    def train_model(self):\n",
+        "        \"\"\"\n",
+        "        Predict target using the target mean.\n",
+        "        \"\"\"\n",
+        "        self.logger.info('Computing mean...')\n",
+        "        self.mean = DummyRegressor(strategy=\"mean\")\n",
+        "        self.mean.fit(self.X_train, self.y_train)\n",
+        "        self.y_pred = self.mean.predict(self.X_test)\n",
```

Comment:
le mean.predict ne fait pas vraiment partie de l'entraînement du modèle, si ?

Reply:
Oui! En fait au début j'entraîne que le modèle dans la méthode "train_model"
mais récemment jai eu un bug, vu que la fonction 'fit' qui vient avant le 'predict' prend un paramètre cat_features différent pour chaque modèle. Donc j'ai du intégrer le 'fit' et le 'predic't dans 'train_model' pour chaque modèle vu que le paramètre est particulier.
Mais effectivement je me suis posée la question : est-ce que je dois renommer la méthode "train_and_predict" ou créer une autre méthode qui fait que le "predict" pour chaque modèle (Single responsibility and all that) ?

Reply:
Finalement, j'ai rajouté une méthode 'predict_target()" dans la class "Model" qui fera ça pour tous les modèles!